modules/dcache/src/main/java/org/dcache/pool/statistics/IoStatistics.java

```diff
@@ -0,0 +1,342 @@
+        _avgRequestedWriteBytes = calculateNewAvg(_writeRequestNum, _avgRequestedWriteBytes, requestedWriteBytes);
+
+        _writeRequestNum++;
+    }
+
+    /**
+     *
+     * Calculate the speed of a request.
+     *
+     * @param  bytes
+     *         The number of processed bytes; must be non-negative
+     *
+     * @param  time
+     *         The duration of the request in milliseconds; must be non-negative
+     *
+     * @return The speed of the process in bytes per seconds.
+     */
+
+    private double calculateSpeed(long bytes, long time){
+        return (double) bytes / ((double) time / TimeUnit.SECONDS.toNanos(1));
+    }
+
+    /**
+     *
+     * Calculate the new average of the speed of all read or write requests.
+     *
+     * @param  requestNum
+     *         The number of read or write requests so far; must be non-negative
+     *
+     * @param  oldAvg
+     *         The old average of the speed of all read or write requests; must be non-negative
+     *
+     * @param  newSpeed
+     *         A new speed value of a read or write request; must be non-negative
+     *
+     * @return The new average of the speed of all read or write requests.
+     */
+
+    private double calculateNewAvg(long requestNum, double oldAvg, double newSpeed){
+        return (requestNum * oldAvg + newSpeed) / (requestNum + 1);
+    }
+
+    public int getReadRequestNum() {
+        return _readRequestNum;
+    }
+
+    public long getTotalReadBytes() {
+        return _totalReadBytes;
+    }
+
+    public double getMinReadSpeed() {
+        return _minReadSpeed;
+    }
+
+    public double getMaxReadSpeed() {
+        return _maxReadSpeed;
+    }
+
+    public double getAvgReadSpeed() {
+        return _avgReadSpeed;
+    }
+
+    public double get95ReadSpeed() {
+        return _95ReadSpeed;
+    }
+
+    public long getTotalReadTime() {
+        return _totalReadTime;
+    }
+
+    public long getTotalRequestedReadBytes() {
+        return _totalRequestedReadBytes;
+    }
+
+    public double getAvgRequestedReadBytes() {
+        return _avgRequestedReadBytes;
+    }
+
+    public int getNegativeReadRequestNum() { return _readRequestNum - _nonNegativeReadRequestNum; }
+
+    public int getWriteRequestNum() {
+        return _writeRequestNum;
+    }
+
+    public long getTotalWrittenBytes() {
+        return _totalWrittenBytes;
+    }
+
+    public double getMinWriteSpeed() {
+        return _minWriteSpeed;
+    }
+
+    public double getMaxWriteSpeed() {
+        return _maxWriteSpeed;
+    }
+
+    public double getAvgWriteSpeed() {
+        return _avgWriteSpeed;
+    }
+
+    public double get95WriteSpeed() {
+        return _95WriteSpeed;
+    }
+
+    public long getTotalWriteTime() {
+        return _totalWriteTime;
+    }
+
+    public long getTotalRequestedWriteBytes() {
+        return _totalRequestedWriteBytes;
+    }
+
+    public double getAvgRequestedWriteBytes() {
+        return _avgRequestedWriteBytes;
+    }
+
```

Comment:
Please add @Override annotation. By this we ensure that mistyped methods names will be catch at compile time.